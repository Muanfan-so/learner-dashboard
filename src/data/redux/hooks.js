import { useSelector } from 'react-redux';

import { actions as appActions } from './app/reducer';
import appSelectors from './app/selectors';
import requestSelectors from './requests/selectors';

const { courseCard } = appSelectors;

export const usePageNumber = () => useSelector(appSelectors.pageNumber);
export const useEmailConfirmationData = () => useSelector(appSelectors.emailConfirmation);
export const useEnterpriseDashboardData = () => useSelector(appSelectors.enterpriseDashboard);
export const usePlatformSettingsData = () => useSelector(appSelectors.platformSettings);
// suggested courses is max at 3 at the moment.
export const useSuggestedCoursesData = () => useSelector(appSelectors.suggestedCourses).slice(0, 3);
export const useRecommendedCoursesData = () => useSelector(appSelectors.recommendedCourses);
export const useSelectSessionModalData = () => useSelector(appSelectors.selectSessionModal);

export const useHasCourses = () => useSelector(appSelectors.hasCourses);
export const useHasAvailableDashboards = () => useSelector(appSelectors.hasAvailableDashboards);
export const useCurrentCourseList = (opts) => useSelector(
  state => appSelectors.currentList(state, opts),
);
export const useShowSelectSessionModal = () => useSelector(appSelectors.showSelectSessionModal);

// eslint-disable-next-line
export const useCourseCardData = (selector) => (cardId) => useSelector(
  (state) => selector(state, cardId),
);

export const useCardCertificateData = useCourseCardData(courseCard.certificate);
export const useCardCourseData = useCourseCardData(courseCard.course);
export const useCardCourseRunData = useCourseCardData(courseCard.courseRun);
export const useCardEnrollmentData = useCourseCardData(courseCard.enrollment);
export const useCardEntitlementData = useCourseCardData(courseCard.entitlement);
export const useCardGradeData = useCourseCardData(courseCard.gradeData);
export const useCardProviderData = useCourseCardData(courseCard.courseProvider);
export const useCardRelatedProgramsData = useCourseCardData(courseCard.relatedPrograms);

export const useUpdateSelectSessionModalCallback = (dispatch, cardId) => () => dispatch(
  appActions.updateSelectSessionModal(cardId),
);

export const useMasqueradeData = () => useSelector(requestSelectors.masquerade);

export const useIsPendingRequest = (requestName) => useSelector(requestSelectors.isPending(requestName));
