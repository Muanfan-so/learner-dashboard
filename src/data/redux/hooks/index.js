import * as app from './app';
import * as requests from './requests';

export * from './app';
export { requests };

export default { app, requests };
