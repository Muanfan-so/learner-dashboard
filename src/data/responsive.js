import { useWindowSize, breakpoints } from '@edx/paragon';
import { StrictDict } from 'utils';

const createResponsive = (fn) => () => {
  const { width } = useWindowSize();
  return fn(width);
};

export const isDesktopSize = createResponsive(
  (width) => width >= breakpoints.large.minWidth,
);

// Beware of using isPadSize() for styling. When the screen resizes pass the
// breakpoint, the screen will flash between the two styles. This is why it is
// recommened isDesktopSize() or isMobileSize() be used instead.
export const isIpadSize = createResponsive(
  (width) => width >= breakpoints.medium.minWidth && width < breakpoints.large.minWidth,
);

export const isMobileSize = createResponsive(
  (width) => width < breakpoints.medium.minWidth,
);

export default StrictDict({
  isDesktopSize,
  isIpadSize,
  isMobileSize,
});
