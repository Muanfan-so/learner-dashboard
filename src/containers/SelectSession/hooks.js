import { hooks as appHooks, actions } from 'data/redux';

import { useDispatch } from 'react-redux';

export const useSelectSession = ({ courseNumber }) => {
  const dispatch = useDispatch();
  const {
    showSessionModal,
    showLeaveSessionInSessionModal,
  } = appHooks.useSelectSessionsModalData();

  const { entitlementSessions } = appHooks.useCardEntitlementsData(courseNumber);

  const { title: courseTitle } = appHooks.useCardCourseData(courseNumber);

  const updateSessionModal = (showModal, showLeaveOption = false) => dispatch(
    actions.app.updateSelectSessionModal({
      showSessionModal: showModal,
      showLeaveSessionInSessionModal: showLeaveOption,
      courseNumber,
    }),
  );

  return {
    showSessionModal,
    closeSessionModal: () => updateSessionModal(false),
    openSessionModal: () => updateSessionModal(true),
    openSessionModalWithLeaveOption: () => updateSessionModal(true, true),
    showLeaveSessionInSessionModal,
    entitlementSessions,
    courseTitle,
  };
};

export default useSelectSession;
