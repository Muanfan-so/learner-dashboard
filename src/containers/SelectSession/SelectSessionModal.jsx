import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import {
  ActionRow, Button, Form, ModalDialog,
} from '@edx/paragon';

import { nullMethod } from 'hooks';
import { dateFormatter } from 'utils';

import useSelectSession from './hooks';
import messages from './messages';

export const SelectSessionModal = ({ courseNumber }) => {
  const {
    entitlementSessions,
    showSessionModal,
    closeSessionModal,
    showLeaveSessionInSessionModal,
    courseTitle,
  } = useSelectSession({
    courseNumber,
  });

  const { formatMessage, formatDate } = useIntl();

  let header;
  let hint;
  if (showLeaveSessionInSessionModal) {
    header = formatMessage(messages.changeOrLeaveHeader);
    hint = formatMessage(messages.changeOrLeaveHint);
  } else {
    header = formatMessage(messages.selectSessionHeader, {
      courseTitle,
    });
    hint = formatMessage(messages.selectSessionHint);
  }

  return (
    <ModalDialog
      isOpen={showSessionModal}
      onClose={nullMethod}
      hasCloseButton={false}
      title=""
    >
      <div
        className="bg-white p-3 rounded shadow"
        size="md"
        style={{ textAlign: 'start' }}
      >
        <h4>{header}</h4>
        <Form.Group>
          <Form.Label>{hint}</Form.Label>
          <Form.RadioSet name="sessions">
            {entitlementSessions?.map((entitle) => (
              <Form.Radio key={entitle.startDate} value={entitle.startDate}>
                {dateFormatter(formatDate, entitle.startDate)} - {dateFormatter(formatDate, entitle.endDate)}
              </Form.Radio>
            ))}
            {showLeaveSessionInSessionModal ? (
              <Form.Radio value="leave">
                {formatMessage(messages.leaveSessionOption)}
              </Form.Radio>
            ) : null}
          </Form.RadioSet>
        </Form.Group>
        <ActionRow>
          <Button variant="tertiary" onClick={closeSessionModal}>
            {formatMessage(messages.nevermind)}
          </Button>
          <Button>{formatMessage(messages.confirmSession)}</Button>
        </ActionRow>
      </div>
    </ModalDialog>
  );
};
SelectSessionModal.propTypes = {
  courseNumber: PropTypes.string.isRequired,
};

export default SelectSessionModal;
