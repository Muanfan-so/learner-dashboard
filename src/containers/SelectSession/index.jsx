import React from 'react';
import { hooks as appHooks } from 'data/redux';

import SelectSessionModal from './SelectSessionModal';

export const SelectSession = () => {
  const { courseNumber } = appHooks.useSelectSessionsModalData();
  return courseNumber ? <SelectSessionModal courseNumber={courseNumber} /> : null;
};
SelectSession.propTypes = {};

export default SelectSession;
