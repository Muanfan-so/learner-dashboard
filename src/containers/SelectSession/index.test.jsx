import React from 'react';
import { shallow } from 'enzyme';

import { hooks as appHooks } from 'data/redux';

import SelectSession from '.';

jest.mock('data/redux', () => ({
  hooks: {
    useSelectSessionsModalData: jest.fn(),
  },
}));

describe('SelectSession', () => {
  describe('snapshot', () => {
    test('no courseNumber', () => {
      appHooks.useSelectSessionsModalData.mockReturnValueOnce({ courseNumber: null });
      expect(shallow(<SelectSession />)).toMatchSnapshot();
    });

    test('has courseNumber', () => {
      appHooks.useSelectSessionsModalData.mockReturnValueOnce({ courseNumber: 'some course' });
      expect(shallow(<SelectSession />)).toMatchSnapshot();
    });
  });
});
