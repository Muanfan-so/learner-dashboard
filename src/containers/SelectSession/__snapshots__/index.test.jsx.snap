// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectSession snapshot has courseNumber 1`] = `
<SelectSessionModal
  courseNumber="some course"
/>
`;

exports[`SelectSession snapshot no courseNumber 1`] = `""`;
